connectionpool.go

```diff
@@ -439,19 +439,14 @@ func (pool *hostConnPool) fill() {
 
 		// filled one
 		fillCount--
-		// connect all remaining connections to this host
```

Comment:
Is there a reason to make this not finish synchronously? The returned session should be fully usable once created but in this case there may have been connection errors which are not caught here. Could you leave this code to return synchronously when the pool is empty to retain previous behaviour.